ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE NumericUnderscores #-}
+
+{-# OPTIONS_GHC "-fno-warn-incomplete-patterns" #-}
+module Test.Ouroboros.Storage.LedgerDB.DiskPolicy (tests) where
+
+import           Data.Function ((&))
+import           Data.Word
+import           Data.Time.Clock (DiffTime, secondsToDiffTime)
+
+import           Test.QuickCheck
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+import           Ouroboros.Consensus.Config.SecurityParam (SecurityParam(..))
+import           Ouroboros.Consensus.Storage.LedgerDB.DiskPolicy (DiskPolicy(..), defaultDiskPolicy, SnapshotInterval(..))
+import           Test.Ouroboros.Storage.LedgerDB.OrphanArbitrary
+
+tests :: TestTree
+tests = testGroup "DiskPolicy/defaultDiskPolicy" [
+      testGroup "onDiskNumSnapshots" [
+          testProperty "should always be equal to 2" prop_onDiskNumSnapshots
+        ]
+    , testGroup "onDiskShouldTakeSnapshot" [
+          testGroup "haven't take a snapshot yet" [
+            testProperty
+              "should take snapshot if it processed at least k blocks"
+              prop_shouldSnapshot_case1
+        ]
+        , testGroup "have previously taken a snapshot" [
+              testProperty
+                "should take snapshot if time since last is greater then explicitly requested interval"
+                prop_shouldSnapshot_case2
+            , testProperty
+                "should take snapshot if time since last is greater then 2 * k if snapshot interval is set to default"
+                prop_shouldSnapshot_case3
+            , testProperty
+                "should take snapshot processed 50k and it's been more then 6 minutes since last snapshot was taken"
+                prop_shouldSnapshot_case4
+        ]
+      ]
+    ]
+
+prop_shouldSnapshot_case1 :: Word64 -> SecurityParam -> SnapshotInterval -> Property
+prop_shouldSnapshot_case1 blocksSinceLast securityParam@(SecurityParam k) snapshotInterval = do
```

Comment:
We mentioned in the call that we should remove these `do` keywords throughout this module.